code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -134,17 +135,61 @@ Implant Specifics:<BR>"}
 <b>Special Features:</b> Explodes<BR>
 <b>Integrity:</b> Implant will occasionally be degraded by the body's immune system and thus will occasionally malfunction."}
 
-/obj/item/implant/dexplosive/trigger(emote, source as mob)
-	if(emote == "deathgasp")
-		src.activate("death")
+/obj/item/implant/explosive/deadman/attack_self(mob/user)
+	return
 
-/obj/item/implant/dexplosive/activate(var/cause)
-	if((!cause) || (!src.imp_in))	return 0
-	explosion(src, -1, 0, 2, 3, 0)//This might be a bit much, dono will have to see.
-	if(src.imp_in)
-		src.imp_in.gib()
+/obj/item/implant/explosive/deadman/hear(var/msg)
+	return
+
+/obj/item/implant/explosive/deadman/process()
+	if(malfunction)
+		STOP_PROCESSING(SSprocessing, src)
+		return
+	if (!implanted)
+		return
+	var/mob/M = imp_in
+
+	if(M.stat == 2)
+		activate("death")
+
+/obj/item/implant/explosive/deadman/activate(var/cause)
+	switch(cause)
+		if("death")
+			small_countdown(src)
+			STOP_PROCESSING(SSprocessing, src)
+
+/obj/item/implant/explosive/deadman/small_boom()
+	if(imp_in)
+		explosion(get_turf(src), 0, 0, 2, 4, 5)
+		qdel(src)
+		imp_in.gib()
+
+/obj/item/implant/explosive/deadman/implanted(mob/source)
+	START_PROCESSING(SSprocessing, src)
+	return 1
+
+/obj/item/implant/explosive/deadman/emp_act(severity)
+	if(malfunction)
+		return
+	malfunction = MALFUNCTION_TEMPORARY
+	switch (severity)
+		if(3.0)
+			if(prob(1))
+				small_countdown()
+			else if(prob(5))
+				meltdown()
+		if(2.0)
+			if(prob(5))
+				small_countdown()
+			else if (prob(10))
+				meltdown()
+		if(1.0)
+			if(prob(10))
+				small_countdown()
+			else if (prob(30))
+				meltdown()
 
-/obj/item/implant/dexplosive/islegal()
+/obj/item/implant/explosive/deadman/islegal()
 	return 0
```

Comment:
if the parent explosive/deadman/islegal() returns false already, you can just get rid of this.